test/integration/features/profile.feature

```diff
@@ -2,104 +2,211 @@
+    Given profile "Test-123" has state "Does Not Exist"
+     When executing "minishift profile set Test-123" succeeds
+     Then profile "Test-123" should be an active profile
+     And executing "minishift profile set minishift" succeeds
+     Then profile "minishift" should be an active profile
+
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is an active profile
+     When executing "minishift start" succeeds
+     Then profile "minishift" should have state "Running"
+     And executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
+
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    Given profile "foo" has state "Does Not Exist"
+     When executing "minishift profile set foo" succeeds
+     Then profile "foo" should be an active profile
+     And executing "minishift config set memory 4096" succeeds
+     And executing "minishift config set disk-size 30g" succeeds
+     And executing "minishift config set cpus 4" succeeds
+     And executing "minishift addons enable registry-route" succeeds
+     Then executing "minishift config get memory" succeeds
+     And stdout should match "4096"
+     Then executing "minishift config get disk-size" succeeds
+     And stdout should match "30g"
+     Then executing "minishift config get cpus" succeeds
+     And stdout should match "4"
+     Then executing "minishift addons list"
+     And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+
+  @minishift-only
+  Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' from active profile 'foo' 
+    Given profile "foo" has state "Does Not Exist"
+    And profile "foo" is an active profile
+     When executing "minishift --profile minishift config get memory" succeeds
+     Then stdout should match "<nil>"
+     When executing "minishift --profile minishift config get disk-size" succeeds
+     Then stdout should match "<nil>"
+     When executing "minishift --profile minishift config get cpus" succeeds
+     Then stdout should match "<nil>"
+     When executing "minishift --profile minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+     And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+     And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+     And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: As a user, I can start profile 'foo'
+    Given profile "foo" has state "Does Not Exist"
+    And profile "foo" is an active profile
+     When executing "minishift start" succeeds
+     Then stdout should contain
      """
-     Default profile 'minishift' can not be deleted
+     Add-on 'registry-route' created docker-registry route.
      """
+     And profile "foo" should have state "Running"
+  
+  @minishift-only
```

Comment:
Do we need to be Minishift specific here. Is the exact match of add-ons important here? Can we not change the regexp in a way that it matches CDK and Minishift. In the add-on tests I would expect that we use @minishift-only, since there we want to assert something which differs. In this feature file we are interested in the profile feature. No need to use @minishift-only